Classes/Private/Libxml2/Data/ElementNode.swift

```diff
@@ -215,6 +366,322 @@ extension Libxml2 {
+                fatalError("This case should not be possible. Review the logic triggering this.")
+            }
+
+            for newNode in newChildren {
+                newNode.parent = self
+            }
+
+            children.removeAtIndex(childIndex)
+            children.insertContentsOf(newChildren, at: childIndex)
+        }
+
+        /// Removes the specified child nodes.  Only updates their parents if specified.
+        ///
+        /// - Parameters:
+        ///     - chldren: the child nodes to remove.
+        ///     - updateParent: whether the children node's parent must be update to `nil` or not.
+        ///             If not specified, the parent is updated.
+        ///
+        func remove(children: [Node], updateParent: Bool = true) {
+            self.children = self.children.filter({ child -> Bool in
+
+                let removeChild = children.contains(child)
+
+                if removeChild && updateParent {
+                    child.parent = nil
+                }
+
+                return !removeChild
+            })
+        }
+
+        /// Retrieves all child nodes positioned after a specified location.
+        ///
+        /// - IMPORTANT: This method can also modify the DOM depending on the value of `splitEdge`.
+        ///         Please refer to the parameter's documentation for more information.
+        ///
+        /// - Parameters:
+        ///     - location: marks the position after which nodes must be positioned.
+        ///     - splitEdge: if this is `true`, any node intersecting `location` will be split
+        ///             so that the part of it after `location` will be returned as a new node.
+        ///             If this is `false`, only nodes that are completely positioned after
+        ///             `location` will be returned and the DOM will be left unmodified.
+        ///
+        /// - Returns: the requested nodes.
+        ///
+        private func children(after location: Int, splitEdge: Bool = false) -> [Node] {
+
+            var result = [Node]()
+            var offset = length()
+
+            for index in (children.count - 1).stride(through: 0, by: -1) {
+
+                let child = children[index]
+
+                offset = offset - child.length()
+
+                let childEndPosition = offset + child.length()
+
+                if offset > location {
+                    result.insert(child, atIndex: 0)
+                } else if splitEdge && childEndPosition > location {
+
+                    let splitRange = NSRange(location: location - offset, length: childEndPosition - location)
+                    child.split(forRange: splitRange)
+
+                    result.insert(child, atIndex: 0)
+                    break
+                } else {
+                    break
+                }
+            }
+            
+            return result
+        }
+
+        /// Retrieves all child nodes positioned before a specified location.
```

Comment:
So in theory it would be either "all children" (without the word "nodes") or "all child nodes".